userbenchmark/cpu/run_config.py

```diff
@@ -0,0 +1,105 @@
+
+"""
+Run PyTorch cpu benchmarking.
+"""
+import argparse
+import os
+import sys
+import numpy
+
+from typing import List, Tuple, Dict, Optional
+from pathlib import Path
+
+REPO_DIR = str(Path(__file__).parent.parent.parent.resolve())
+sys.path.append(REPO_DIR)
+
+from utils import add_path
+with add_path(REPO_DIR):
+    from userbenchmark.cpu.cpu_utils import REPO_PATH, get_output_json, dump_output
+    from torchbenchmark.util.experiment.instantiator import list_models, load_model_isolated, TorchBenchModelConfig, \
+                                                            list_devices, list_tests
+    from torchbenchmark.util.experiment.metrics import TorchBenchModelMetrics, get_model_test_metrics
+
+BM_NAME = 'cpu'
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+def get_metrics(_config: TorchBenchModelConfig) -> List[str]:
+    return ["latencies", "cpu_peak_mem"]
+
+def get_output_subdir(config: TorchBenchModelConfig) -> str:
+    mode = "jit" if config.jit else "eager"
+    subdir = f"{config.test}_{config.name}_{mode}"
+    return subdir
+
+def result_to_output_metrics(metrics: TorchBenchModelMetrics) -> Dict[str, float]:
+    result_metrics = {}
+    if metrics.latencies:
+        latency_metric = "latency"
+        median_latency = numpy.median(metrics.latencies)
+        assert median_latency, f"Run failed for metric {latency_metric}"
+        result_metrics[latency_metric] = median_latency
+    if metrics.cpu_peak_mem:
+        cpu_peak_mem = "cpu_peak_mem"
+        result_metrics[cpu_peak_mem] = metrics.cpu_peak_mem
+    return result_metrics
+
+def dump_result_to_json(metrics, output_dir):
+    result = get_output_json(BM_NAME, metrics)
+    dump_output(BM_NAME, result, output_dir)
+
+def validate(candidate: str, choices: List[str]) -> str:
+    """Validate the candidates provided by the user is valid"""
+    assert candidate in choices, f"Specified {candidate}, but not in available list: {choices}."
+    return candidate
+
+def run_config(config: TorchBenchModelConfig, dryrun: bool=False) -> Optional[TorchBenchModelMetrics]:
+    """This function only handles NotImplementedError, all other errors will fail."""
+    metrics = get_metrics(config)
+    print(f"Running {config} ...", end='')
+    if dryrun:
+        return None
+    # We do not allow RuntimeError in this test
+    try:
+        # load the model instance within the same process
```

Comment:
To clarify: `load_model_isolated()` loads model in separate subprocesses.

Reply:
Is there any potential issue in here? I have double checked the numactl setting should be apply to subprocess also, am I right?

Reply:
Running multiple models within the same process is known to be problematic on GPU (for example, it won't release GPU memory after running.), but I am not sure if it is the same case on CPU.
Here I am just saying if you decide to run in subprocesses, the comment text needs to be updated to avoid confusion.

Reply:
Got you. Thanks, will update the comment text firstly